tests/tracking/test_model_registry.py

```diff
@@ -386,6 +299,40 @@ def test_update_create_model_version_flow(mlflow_client, backend_store_uri):
     assert_is_between(start_time_2, end_time_2, rmd4.last_updated_timestamp)
 
 
+def test_latest_models(mlflow_client, backend_store_uri):
+    version_stage_mapping = {
+        1: "Archived",
+        2: "Production",
+        3: "Archived",
+        4: "Production",
+        5: "Staging",
+        6: "Staging",
+        7: "None",
+    }
+    name = 'LatestVersionTest'
+    mlflow_client.create_registered_model(name)
+
+    version_mvd_mapping = {}
+    for version, stage in version_stage_mapping.items():
+        mv = mlflow_client.create_model_version(name, "path/to/model", "run_id")
+        assert mv.version == version
+        if stage != "None":
+            mlflow_client.update_model_version(name, version, stage=stage)
+        mvd = mlflow_client.get_model_version_details(name, version)
+        assert mvd.current_stage == stage
+        version_stage_mapping[version] = mvd
```

Comment:
do you need this? seems like we are unnecessarily modifying the map in flight?

Reply:
you are right! Not needed anymore...